lib/broadway/producer.ex

```diff
@@ -47,11 +47,20 @@ defmodule Broadway.Producer do
   def init({args, index}) do
     {module, arg} = args[:module]
     transformer = args[:transformer]
+    n_processors = args[:n_processors]
+    processors_partition_by = args[:processors_partition_by]
+    processors_name = args[:processors_name]
+    batchers_names = args[:batchers_names]
+    broadway_name = args[:broadway_name]
 
     # Inject the topology index only if the args are a keyword list.
     arg =
       if Keyword.keyword?(arg) do
-        Keyword.put(arg, :broadway_index, index - 1)
+        arg
+        |> Keyword.put(:broadway_index, index - 1)
+        |> Keyword.put(:processors_name, processors_name)
```

Comment:
We should make `processor_names` and `batcher_names` consistent with `producer_names` (singular and then plural).